packages/components/src/List/ListComposition/ColumnChooser/ColumnChooser.component.js

```diff
@@ -20,7 +21,7 @@ function ColumnChooser(props) {
 					props.onSubmit(_, changes);
 				}
 			}}
-			{...props}
+			{...omit(props, 'onSubmit')}
```

Comment:
What you want is to spread props before the onSubmit I guess
```suggestion
			{...props}
			onSubmit={(_, changes) => {
				setVisibleColumns(changes.filter(c => !c.hidden).map(c => c.key));
				if (props.onSubmit) {
					props.onSubmit(_, changes);
				}
			}}
```